segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/SegmentMetadataUpdateTransaction.java

```diff
@@ -478,6 +480,9 @@ private void preProcessAttributes(AttributeUpdateCollection attributeUpdates)
             if (Attributes.isUnmodifiable(u.getAttributeId())) {
                 throw new MetadataUpdateException(this.containerId,
                         String.format("Attribute Id '%s' on Segment Id %s ('%s') may not be modified.", u.getAttributeId(), this.id, this.name));
+            } else if (NameUtils.isTransientSegment(this.name) && !Attributes.isCoreAttribute(u.getAttributeId())) {
```

Comment:
How come? We should allow extended attributes on transient segments. We should just make sure their total count doesn't exceed some small threshold we set.

Reply:
Was working on the changes after I submitted the PR.. fixing.